simplesat/repository.py

```diff
@@ -75,7 +75,7 @@ def add_package(self, package_metadata):
         If the same package is added multiple times to a repository, every copy
         will be available when calling find_package or when iterating.
         """
-        if package_metadata.name not in self._name_to_packages:
+        if len(self._name_to_packages[package_metadata.name]) == 0:
```

Comment:
I converted to using a normal dict to see how painful it would be, and I think it makes the code clearer anyway :)